NSFS | NC | nsfs service additions

romayalon opened the pull request:
### Explain the changes
1. Added needed env vars to nsfs service
2. Added a check when creating fs_napi for GPFS lib file existence before reading the lib file - and added this check boolean in use_gpfs_lib().

### Issues: Fixed #xxx / Gap #xxx
1. https://issues.redhat.com/browse/MCGI-201

### Testing Instructions:
1. Run NC nsfs with the GPFS path as env variable while lib file exists
2. Run NC nsfs with the GPFS path as env variable while lib file does not exist
3. Run NC nsfs without the GPFS path as env variable

- [ ] Doc added/updated
- [ ] Tests added

romayalon committed:
gpfs load if lib exists

Signed-off-by: Romy <35330373+romayalon@users.noreply.github.com>

tangledbytes commented on src/deploy/nsfs.service:
```diff
@@ -1,12 +1,15 @@
 [Unit]
 Description=The NooBaa nsfs service.
+After=gpfs-wait-mount.service
```
Will this prevent the service to start if `gpfs-wait-mount.service` does not exist?

romayalon replied:
no, it will work event without it

tangledbytes commented on src/deploy/nsfs.service:
```diff
@@ -1,12 +1,15 @@
 User=root
 Group=root
-Environment=
+Environment=ENDPOINT_FORKS=0
```
Should we also allow another place to configure this, like `system.json` (eg. `"forks": 8`) so that consumers of this service don't have to edit the unit file to configure the forks?

romayalon replied:
I think that we can specify an .env file instead under the config dir, WDYT?

tangledbytes replied:
On a second thought, I think `config-local.js` is good enough for that purpose.

tangledbytes replied:
> I think that we can specify a .env file instead under the config dir, WDYT?
Makes sense!

romayalon commented:
@tangledbytes I added a default nsfs_env.env file as an example that should be linked to /etc/noobaa.conf.d/.env
and added it as an environmentFile to be used in nsfs.service, please re-check it

tangledbytes commented on src/deploy/nsfs.service:
```diff
@@ -1,13 +1,14 @@
 Group=root
-Environment=
 ExecStart=/usr/local/noobaa-core/bin/node /usr/local/noobaa-core/src/cmd/nsfs.js
+EnvironmentFile=/etc/noobaa.conf.d/.env
```
We already have support for `$NOOBAA_DIR/.env` file though. Should we document somewhere that these two `.env` files are different and that `$NOOBAA_DIR/.env` will always supersede `/etc/noobaa.conf.d/.env` one?
